homeassistant/components/dsmr/sensor.py

```diff
@@ -32,47 +32,51 @@
 ICON_POWER_FAILURE = "mdi:flash-off"
 ICON_SWELL_SAG = "mdi:pulse"
 
-# Smart meter sends telegram every 10 seconds
-MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
-
 RECONNECT_INTERVAL = 5
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
         vol.Optional(CONF_HOST): cv.string,
         vol.Optional(CONF_DSMR_VERSION, default=DEFAULT_DSMR_VERSION): vol.All(
-            cv.string, vol.In(["5", "4", "2.2"])
+            cv.string, vol.In(["5B", "5", "4", "2.2"])
         ),
         vol.Optional(CONF_RECONNECT_INTERVAL, default=30): int,
         vol.Optional(CONF_PRECISION, default=DEFAULT_PRECISION): vol.Coerce(int),
     }
 )
 
+# Smart meter sends telegram every 10 seconds or every second for DSMR 5
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=1)
+
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the DSMR sensor."""
     # Suppress logging
     logging.getLogger("dsmr_parser").setLevel(logging.ERROR)
 
+    global MIN_TIME_BETWEEN_UPDATES
```

Comment:
No `global` please.

Reply:
Well I needed the global scope cause I wanted to update the MIN_TIME_BETWEEN_UPDATES depending on the configured dsmr_version...
Any proposals how I could improve this?